### PR TITLE
Add validator for SyncCommitteeSignature

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Add validator for `SyncCommitteeSignature` gossip.

Uses `ValidateableSyncCommitteeSignature` similar to `ValidateableAttestation` to wrap the actual gossip message with other useful and cached info like the subnet it was received on.

## Fixed Issue(s)
#3803 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/versions/altair/ValidateableSyncCommitteeSignature.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  public Set<Integer> calculateApplicableSubcommittees(final Spec spec, final BeaconState state) {
+    final Optional<Set<Integer>> currentValue = this.applicableSubcommittees;
+    if (currentValue.isPresent()) {
+      return currentValue.get();
+    }
+    final Set<Integer> applicableSubcommittees =
+        spec.getSyncCommitteeUtil(signature.getSlot())
+            .orElseThrow()
```
Comment (mbaxter) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/versions/altair/ValidateableSyncCommitteeSignature.java:64]: Might be worth a custom error here?

Comment (mbaxter) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/versions/altair/ValidateableSyncCommitteeSignature.java]: Might be worth a custom error here?  And maybe standardizing it with `spec.getRequiredSyncCommitteeUtil()`.

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeSignatureValidator.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    final UniquenessKey uniquenessKey = getUniquenessKey(signature);
+    if (seenIndices.contains(uniquenessKey)) {
+      return SafeFuture.completedFuture(IGNORE);
+    }
+
+    // [IGNORE] The block being signed over (sync_committee_signature.beacon_block_root) has been
+    // seen (via both gossip and non-gossip sources).
+    if (!recentChainData.containsBlock(signature.getBeaconBlockRoot())) {
+      LOG.trace("Ignoring sync committee signature because beacon block is not known");
+      return SafeFuture.completedFuture(IGNORE);
```
Comment (mbaxter) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeSignatureValidator.java:91]: Should we be saving this?:
```suggestion
      return SafeFuture.completedFuture(SAVE_FOR_FUTURE);
```

**Review by mbaxter — APPROVED** [review]
LGTM

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/versions/altair/ValidateableSyncCommitteeSignature.java (hunk 1 above)
Comment (ajsutton) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/versions/altair/ValidateableSyncCommitteeSignature.java]: Done

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeSignatureValidator.java (hunk 2 above)
Comment (ajsutton) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeSignatureValidator.java:91]: There's no provision in the network spec to save these for the future.  I'm not sure there's much value in adding the complexity of queuing given they're only valid for one slot - if the block is late all the votes will likely be for the previous block anyway and will be useless by the time we get the next block.

**Pull request merged**